ui/tests/integration/components/keymgmt/distribute-test.js

```diff
@@ -78,33 +78,53 @@ module('Integration | Component | keymgmt/distribute', function (hooks) {
           }),
         ];
       });
+      this.get('/v1/keymgmt/kms', (response) => {
+        return [
+          response,
+          { 'Content-Type': 'application/json' },
+          JSON.stringify({
+            data: {
+              keys: ['provider-aws', 'provider-azure', 'provider-gcp'],
+            },
+          }),
+        ];
+      });
     });
   });
 
   hooks.afterEach(function () {
     this.server.shutdown();
   });
 
-  test('it does not allow operation selection until valid key and provider selected', async function (assert) {
+  test('it does not allow operation selection until valid key/provider combo selected', async function (assert) {
+    assert.expect(6);
     await render(
-      hbs`<Keymgmt::Distribute @backend="keymgmt" @providers={{providers}} @onClose={{fn (mut this.onClose)}} />`
+      hbs`<Keymgmt::Distribute @backend="keymgmt" @key="example-1" @providers={{providers}} @onClose={{fn (mut this.onClose)}} />`
     );
     assert.dom(SELECTORS.operationsSection).hasAttribute('disabled');
+    // Select
     await clickTrigger();
     await settled();
-    assert.equal(ssComponent.options.length, 3, 'shows all key options');
+    assert.equal(ssComponent.options.length, 3, 'shows all provider options');
+    await typeInSearch('aws');
     await ssComponent.selectOption();
     await settled();
-    assert.dom(SELECTORS.operationsSection).hasAttribute('disabled');
-    await select(SELECTORS.providerInput, 'provider-aws');
-    await settled();
     assert.dom(SELECTORS.operationsSection).doesNotHaveAttribute('disabled');
-    await select(SELECTORS.providerInput, 'provider-azure');
+    // Remove selection
+    await ssComponent.deleteButtons.objectAt(0).click();
+    // Select Azure
+    await settled();
+    await clickTrigger();
+    await settled();
```

Comment:
I wonder if `settled` needs to be called after `clickTrigger`? I had a look and that helper calls `click` from _@ember/test-helpers_ which calls settled within that method.

Reply:
I got failures that such-and-such didn't exist when I removed `settled` but I can try again

Reply:
Update: removed settled and it worked fine! 🎉